Source/WebKit/UIProcess/Cocoa/SOAuthorization/RedirectSOAuthorizationSession.mm

```diff
@@ -67,34 +67,22 @@
     auto* navigationAction = this->navigationAction();
     ASSERT(navigationAction);
     auto* page = this->page();
-    if ((response.httpStatusCode() != 302 && response.httpStatusCode() != 200) || !page) {
+    if ((!response.isRedirection() && response.httpStatusCode() != 200) || !page) {
         AUTHORIZATIONSESSION_RELEASE_LOG("completeInternal: httpState=%d page=%d, so falling back to web path.", response.httpStatusCode(), !!page);
         fallBackToWebPathInternal();
         return;
     }
-    invokeCallback(true);
-    if (response.httpStatusCode() == 302) {
-#if PLATFORM(IOS)
-        // MobileSafari has a WBSURLSpoofingMitigator, which will not display the provisional URL for navigations without user gestures.
-        // For slow loads that are initiated from the MobileSafari Favorites screen, the aforementioned behavior will create a period
-        // after authentication completion where the new request to the application site loads with a blank URL and blank page. To
-        // workaround this issue, we load an html page that does a client side redirection to the application site on behalf of the
-        // request URL, instead of directly loading a new request. The html page should be super fast to load and therefore will not
-        // show an empty URL or a blank page. These changes ensure a relevant URL bar and useful page content during the load.
-        if (!navigationAction->isProcessingUserGesture()) {
```

Comment:
Your comments don't explain why you feel that it is safe to remove this logic. I guess the idea is that we don't fire the redirect at all anymore, we just register the redirect then proceed so that the redirect will get handled as the next step. So therefore we don't need to invoke the spoof mitigation.

I'm worried we still need the Spoof mitigation; I wonder if you could build the data URL in a new request, and register *that* before failing the callback?

Reply:
I am not sure either how to test it.
What happens now with the patch is as if the load proceeds and there is a server side redirection.
The spoofing mitigator should already be handling correctly server side redirections (probably not do anything).